### PR TITLE
Iss hipcms 522 analysis

### PR DESCRIPTION
**Pull request by maltehol**

### CODE REVIEW

File: src/Api/Controllers/TopicsController.Analytics.cs [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+        [ProducesResponseType(typeof(void), 404)]
+        [ProducesResponseType(typeof(void), 403)]
+        public IActionResult GetTagFrequencyAnalytics([FromRoute]int topicId)
+        {
+            if (!_topicPermissions.IsAssociatedTo(User.Identity.GetUserId(), topicId))
+                return Forbidden();
+
+            try
+            {
+                return Ok(new TagFrequencyAnalyticsResult() {TagFrequency = _analyticsManager.GetFrequencyAnalytic(topicId) });
```
Comment (sstamm) [inline src/Api/Controllers/TopicsController.Analytics.cs]: The GetFrequencyAnalytic can already return TagFrequencyAnalyticsResult.

File: src/Api/Managers/DocumentManager.cs [hunk 2]
```diff
@@ -42,6 +49,45 @@ internal EntityResult UpdateDocument(int topicId, int userId, string htmlContent
...
+                        IdInDocument = tagInstanceId,
+                        Value = tagValue,
+                        PositionInDocument = tagPosition,
+                        InDocument = document
+                    };
+
+                    tagInstances.Add(tag);
+                }
+                var oldInstances = DbContext.AnnotationTagInstances;
+                DbContext.AnnotationTagInstances.RemoveRange(oldInstances);
```
Comment (sstamm) [inline src/Api/Managers/DocumentManager.cs]: I think the Topic/Document Id is missing. Here you delete all Instanced inside the database.

File: src/Api/Models/Entity/Annotation/AnnotationTagInstance.cs [hunk 3]
```diff
@@ -14,6 +14,14 @@ public class AnnotationTagInstance
 
         public AnnotationTag TagModel { get; set; }
 
+        public string Value { get; set; }
+
+        public Document InDocument { get; set; }
+
+        public int IdInDocument { get; set; }
```
Comment (sstamm) [inline src/Api/Models/Entity/Annotation/AnnotationTagInstance.cs]: To be consistent it should named InDocoumentId.

File: src/Api/Models/Entity/Annotation/AnnotationTagInstance.cs [hunk 4]
```diff
@@ -28,8 +36,12 @@ public class AnnotationTagInstanceMap
         {
             public AnnotationTagInstanceMap(EntityTypeBuilder<AnnotationTagInstance> entityBuilder)
             {
-                entityBuilder.HasOne(tag => tag.TagModel).WithMany(model => model.TagInstances)
-                    .HasForeignKey(tag => tag.TagModelId).OnDelete(DeleteBehavior.Cascade);
+                entityBuilder.HasOne(tag => tag.TagModel)
+                    .WithMany(model => model.TagInstances)
+                    .HasForeignKey(tag => tag.TagModelId)
+                    .OnDelete(DeleteBehavior.Cascade);
+                entityBuilder.HasOne(tag => tag.InDocument).WithMany()
```
Comment (sstamm) [inline src/Api/Models/Entity/Annotation/AnnotationTagInstance.cs]: HasForeignKey is missing

entityBuilder.HasOne(tag => tag.InDocument).WithMany()**.HasForeignKey(tag => tag.InDocumentId)**.OnDelete(DeleteBehavior.Cascade)

File: src/Api/Managers/DocumentManager.cs (hunk 2 above)
Comment (maltehol) [inline src/Api/Managers/DocumentManager.cs]: head -> table

**Pull request merged**